autodiff.py

```
@@ -1024,6 +1050,7 @@ def jacobian(self, output_jacobian):
         ]
 
     def s2s_expr(self, inputs):
+        # TODO: currently it doesn't support sparse format
```

It would be better to raise an exception when the self.format is not dense.